internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,279 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+		if err := function.CreateRetryGenerateName(r.Client, ctx, deleteBackupRequest); err != nil {
+			logger.Error(err, "Failed to create delete request for VeleroBackup", "BackupName", veleroBackup.Name)
+			return false, err
+		}
+		logger.V(1).Info("Request to delete backup submitted successfully", "BackupName", veleroBackup.Name)
+		nab.Status.VeleroDeleteBackupRequest = &nacv1alpha1.VeleroDeleteBackupRequest{
+			NameUUID:  veleroBackupNameUUID,
+			Namespace: r.OADPNamespace,
+			Name:      deleteBackupRequest.Name,
+		}
+	}
+	// Ensure that the NonAdminBackup's NonAdminBackupStatus is in sync
+	// with the DeleteBackupRequest. Any required updates to the NonAdminBackup
+	// Status will be applied based on the current state of the DeleteBackupRequest.
+	updated := updateNonAdminBackupDeleteBackupRequestStatus(&nab.Status, deleteBackupRequest)
+	if updated {
+		if err := r.Status().Update(ctx, nab); err != nil {
+			logger.Error(err, "Failed to update NonAdminBackup Status after DeleteBackupRequest reconciliation")
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackup DeleteBackupRequest Status updated successfully")
+	}
+	return false, nil // Continue so initNabDeletion can initialize deletion of a NonAdminBackup object
+}
+
+// deleteVeleroBackupAndDeleteBackupRequestObjects deletes both the VeleroBackup and any associated
+// DeleteBackupRequest objects for a given NonAdminBackup when force deletion is requested.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime
+//   - logger: Logger instance
+//   - nab: NonAdminBackup object
+//
+// Returns:
+//   - bool: whether to requeue (always false)
+//   - error: any error encountered during deletion
+func (r *NonAdminBackupReconciler) deleteVeleroBackupAndDeleteBackupRequestObjects(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if nab.Status.VeleroBackup == nil || nab.Status.VeleroBackup.NameUUID == constant.EmptyString {
+		return false, nil
+	}
+
+	veleroBackupNameUUID := nab.Status.VeleroBackup.NameUUID
+	veleroBackup, err := function.GetVeleroBackupByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
+
+	if err != nil {
+		// Case where more than one VeleroBackup is found with the same label UUID
+		// TODO: Determine if all objects with this UUID should be deleted
+		logger.Error(err, findSingleVBError, uuidString, veleroBackupNameUUID)
+		return false, err
+	}
+
+	if veleroBackup != nil {
+		if err = r.Delete(ctx, veleroBackup); err != nil {
+			logger.Error(err, "Failed to delete VeleroBackup", nameString, veleroBackup.Name)
+			return false, err
+		}
+		logger.V(1).Info("VeleroBackup deletion initiated", nameString, veleroBackup.Name)
+	} else {
+		logger.V(1).Info("VeleroBackup already deleted")
+	}
+
+	deleteBackupRequest, err := function.GetVeleroDeleteBackupRequestByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
```

Comment:
since DeleteBackupRequest controller will, eventually, do this clean up for us, would let it do the deletion for us

Reply:
@shubham-pampattiwar please comment on this.

Reply:
Yeah Velero eventually cleans up the residual DBR. But there is no harm in deleting the DRBs ourselves right ? I think its a good step to cleanup DBR objects that we created for the particular NAB and VB and not keep them dangling in cluster.